qiskit_machine_learning/algorithms/classifiers/neural_network_classifier.py

```diff
@@ -174,4 +185,65 @@ def _validate_input(self, X: np.ndarray, y: np.ndarray = None) -> Tuple[np.ndarr
             elif scipy.sparse.issparse(y):
                 y = cast(spmatrix, y).toarray()  # cast is required by mypy
 
-        return X, y
+        return np.asarray(X), np.asarray(y)
```

Comment:
May be I missed this in the previous review, but why `np.asarray()` is requried? Is this caused by pylint/mypy/other problems?

Reply:
So, I think this was required by mypy, but that was when the fix for #320 was part of this PR. It might not be strictly necessary now. I will check.